### DISCUSSION TITLE
Difference between poppers and portals on Menu

### DISCUSSION THREAD

**Discussion by ngtrthinh169TPC** [radix-ui/primitives#2426] [Help]
I'm trying some of Radix's components and reading their docs (particularly the DropdownMenu).

From what I can see from the docs of [DropdownMenu from `Shadcn/UI`](https://ui.shadcn.com/docs/components/dropdown-menu) (a collection of components using Radix), they just omit the Portal and the dropdown works normally.

I can't see any differences between using DropdownMenu.Portal and omitting it since both cases the Content part are both being put to document.body. So I'd be thankful if you guys can show me the differences.

**Comment by benoitgrelard** [comment]
FWIW they do use the portal part on the link you provided.
With or without portal, in most cases it will all look the same.
Portal just enabled the content to be appended to the body rather than right next to the trigger in situ.
This is better for layering, allowing you to break out of clipping contexts, etc
